### PR TITLE
Convert to EF Core 3 (partially)

### PR DESCRIPTION
**Pull request by miniyou**
This PR adds basic support for EF Core 3. Batch delete and update is unsupported, due to the new SQL generated by EF Core.

### CODE REVIEW

File: EntityFramework.Utilities/EntityFramework.Utilities.SqlServer/SqlQueryAnalyzer.cs [hunk 1]
```diff
@@ -30,26 +32,25 @@ public class SqlQueryAnalyzer
...
+
+            var sqlGenerator = factory.Create();
+            var command = sqlGenerator.GetCommand(selectExpression);
 
+            string sql = command.CommandText;
             return sql;
         }
     }
+
+    internal static class ReflectionExtensions
```
Comment (miniyou) [inline EntityFramework.Utilities/EntityFramework.Utilities.SqlServer/SqlQueryAnalyzer.cs:51]: Not the nicest, I just put it here to get it working.

File: EntityFramework.Utilities/EntityFramework.Utilities.SqlServer/SqlServerBatchOperation.cs [hunk 2]
```diff
@@ -29,8 +29,9 @@ public interface ISqlServerBatchOperationBase<TContext, T> where T : class
 
     public interface ISqlServerBatchOperationFiltered<TContext, T>
     {
-        Task<int> DeleteAsync(SqlServerDeleteSettings settings = null);
-        Task<int> UpdateAsync<TP>(Expression<Func<T, TP>> prop, Expression<Func<T, TP>> modifier, SqlServerUpdateSettings settings = null);
+        // // TODO: Full EF Core 3 support
```
Comment (miniyou) [inline EntityFramework.Utilities/EntityFramework.Utilities.SqlServer/SqlServerBatchOperation.cs:33]: FYI: I left this comment on every place where I disabled support for batch delete and update.

File: EntityFramework.Utilities/EntityFramework.Utilities/MappingHelper.cs [hunk 3]
```diff
@@ -170,16 +170,16 @@ public EfMapping(DbContext db)
...
-                        ColumnName = p.SqlServer().ColumnName,
+                        ColumnName = p.GetColumnName(),
                         PropertyName = string.IsNullOrWhiteSpace(basePath) ? p.Name : basePath + "." + p.Name,
-                        DataType = p.SqlServer().ColumnType,
+                        DataType = p.GetColumnType(),
                         ForEntityType = clrType, // Use to get the right type for the whole tree 
                         IsPrimaryKey = p.IsPrimaryKey(),
-                        IsStoreGenerated = p.SqlServer().ValueGenerationStrategy != null, // Can be null | HiLo | Identity. Only null is not generated
-                        DataTypeFull = p.SqlServer().ColumnType,
+                        IsStoreGenerated = p.GetValueGenerationStrategy() != SqlServerValueGenerationStrategy.None,
```
Comment (miniyou) [inline EntityFramework.Utilities/EntityFramework.Utilities/MappingHelper.cs:181]: I don't really know what this means, but comparing with `SqlServerValueGenerationStrategy.None` made it work.

File: EntityFramework.Utilities/Tests/connectionStrings.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 ﻿{
-	"SqlServer" : "Data Source=./; Initial Catalog=BatchTests; Integrated Security=SSPI; MultipleActiveResultSets=True",
-	"SqlCe" : "Data Source=BatchTests.sdf"
+	"SqlServer": "Server=(localdb)\\MSSQLLocalDB;Integrated Security=true; MultipleActiveResultSets=True",
```
Comment (miniyou) [inline EntityFramework.Utilities/Tests/connectionStrings.json]: Oops, committed by accident. Also, this isn't used (rather, the connection string is hardcoded in three different contexts.

```suggestion
	"SqlServer" : "Data Source=./; Initial Catalog=BatchTests; Integrated Security=SSPI; MultipleActiveResultSets=True",
```

**Comment by RudeySH** [comment]
> Batch delete and update is unsupported, due to the new SQL generated by EF Core.

Is it impossible in EF Core to implement it? Or is it unsupported because it's a lot of work to rewrite the existing code to make it work?

**Comment by miniyou** [comment]
It should absolutely be possible to get working, but as an initial step it's fine for us to get inserts only working. We're relying on this library in production, and this upgrade is necessary to unblock upgrading the rest of the code to EF Core 3 (it's blocking due to the EF Core change from `System.Data.SqlClient` to `Microsoft.Data.SqlClient`).

When we're not blocked on this, it's up to Mikael to decide whether it's worth the effort to add support for updates and deletes. Honestly, the bulk of that job would probably not be the implementation _per se,_ but rather to ensure that it's safe in all corner cases.

File: EntityFramework.Utilities/EntityFramework.Utilities/MappingHelper.cs (hunk 3 above)
Comment (MikaelEliasson) [reply EntityFramework.Utilities/EntityFramework.Utilities/MappingHelper.cs:181]: Should be the same. It's if the value is generated by SQL or not. Typical case is an int PK column with an identity seed.

**Comment by MikaelEliasson** [comment]
We will have to look at how easy the new query generation in EF Core 3 is to support. Hopefully it's possible to generate the queries needed instead of parsing them.

This branch is quite hidden and it's basically only we at Bokio that use it so far looking at the download stats. 

https://www.nuget.org/packages/ME.EntityFramework.Utilities

**Pull request merged**